src/components/DataSampleGridView.tsx

```diff
@@ -56,37 +76,78 @@ const DataSamplesGridView = () => {
 
   return (
     <>
-      <RecordingDialog
-        gestureId={selectedGesture.ID}
-        isOpen={isOpen}
-        onClose={onClose}
-        actionName={selectedGesture.name}
+      <ConnectToRecordDialog
+        isOpen={connectToRecordDialogDisclosure.isOpen}
+        onClose={connectToRecordDialogDisclosure.onClose}
       />
+      {selectedGesture && (
+        <RecordingDialog
+          gestureId={selectedGesture.ID}
+          isOpen={isOpen}
+          onClose={onClose}
+          actionName={selectedGesture.name}
+        />
+      )}
       <HeadingGrid
         position="sticky"
         top={0}
         {...gridCommonProps}
         headings={headings}
       />
-      <Grid
-        {...gridCommonProps}
-        alignItems="start"
-        autoRows="max-content"
-        overflow="auto"
-        flexGrow={1}
-        h={0}
-      >
-        {gestures.map((g, idx) => (
-          <DataSampleGridRow
-            key={g.ID}
-            gesture={g}
-            selected={selectedGesture.ID === g.ID}
-            onSelectRow={() => setSelectedGestureIdx(idx)}
-            startRecording={onOpen}
-            showWalkThrough={showWalkThrough}
-          />
-        ))}
-      </Grid>
+      {gestures.length === 0 ? (
+        <VStack
+          gap={5}
+          flexGrow={1}
+          alignItems="center"
+          justifyContent="center"
+        >
+          <LoadProjectInput ref={loadProjectInputRef} accept=".json" />
+          <Text fontSize="lg">No data samples</Text>
```

Comment:
This and the copy below needs to be extracted for translation

Reply:
[d863bf2](https://github.com/microbit-foundation/ml-trainer/pull/331/commits/d863bf2e4d9beacfe922fcc205dd4ac177004988)

Reply:
Oops, I didn't read the message properly and missed "the copy below needs to be extracted for translation"